File: hackerrank/day14/main.cs
using System;

class Difference
{
    private int[] elements;
    public int maximumDifference;

    public Difference(int[] a)
    {
        elements = a;
        maximumDifference = 0;
    }

    public void computeDifference()
    {
        int maximo = elements[0];
        int minimo = elements[0];

        foreach (int numero in elements)
        {
            if (numero > maximo)
            {
                maximo = numero;
            }

            if (numero < minimo)
            {
                minimo = numero;
            }
        }

        maximumDifference = maximo - minimo;
    }
}

class Program
{
    static void Main(string[] args)
    {
        Console.Write("¿Cuántos números desea ingresar? ");
        int n = int.Parse(Console.ReadLine());

        int[] numeros = new int[n];

        for (int i = 0; i < n; i++)
        {
            Console.Write($"Número #{i + 1}: ");
            numeros[i] = int.Parse(Console.ReadLine());
        }

        Difference d = new Difference(numeros);

        d.computeDifference();

        Console.WriteLine("\nDiferencia máxima: " + d.maximumDifference);
    }
}

File: hackerrank/day14/main.js
'use strict';

const readline = require('readline');

class Difference {
    constructor(a) {
        this.elements = a;
        this.maximumDifference = 0;
    }

    computeDifference() {
        let maximo = this.elements[0];
        let minimo = this.elements[0];

        for (let numero of this.elements) {
            if (numero > maximo) {
                maximo = numero;
            }

            if (numero < minimo) {
                minimo = numero;
            }
        }

        this.maximumDifference = maximo - minimo;
    }
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('¿Cuántos números desea ingresar? ', (respuesta) => {

    const n = parseInt(respuesta);
    let numeros = [];
    let contador = 0;

    function pedirNumero() {
        if (contador === n) {

            const d = new Difference(numeros);

            d.computeDifference();

            console.log('\nDiferencia máxima: ' + d.maximumDifference);

            rl.close();
            return;
        }

        rl.question(`Número #${contador + 1}: `, (respuesta) => {
            numeros.push(parseInt(respuesta));
            contador++;

            pedirNumero();
        });
    }

    pedirNumero();
});
